packages/react/src/auth/useSignOut.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useContext, useEffect } from "react";
+import { GadgetConfigurationContext, useApi } from "../../src/GadgetProvider";
+import { useAction } from "../../src/useAction";
+import { useUser } from "./useUser";
+
+/**
+ * Returns a callback that will call the configured `signOutActionApiIdentifier` on the `User` model and optionally redirect (by default).
+ * @returns The `Promise` to call to execute the `signOut` action. Will automatically redirect to the configured `signInPath` unless the `redirect: false` option is provided.
+ */
+export const useSignOut = (opts?: { redirectOnSuccess?: boolean; redirectToPath?: string }) => {
+  const redirectOnSuccess = opts?.redirectOnSuccess ?? true;
+  const redirectToPath = opts?.redirectToPath;
+  const api = useApi();
+  const user = useUser();
+  const context = useContext(GadgetConfigurationContext);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { signOutActionApiIdentifier, signInPath } = context!.auth;
+  if (signOutActionApiIdentifier && (api as any).user[signOutActionApiIdentifier]) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const [{ fetching, error, data }, signOutAction] = useAction((api as any).user[signOutActionApiIdentifier]);
```

Comment:
will the user need to hook into when the sign out action is running? maybe we should return fetching or enable suspense here